src/openloco/windows/station_list.cpp

```diff
@@ -17,36 +37,739 @@ namespace openloco::ui::windows::station_list
+                ptr = stringmgr::format_string(ptr, objectmgr::get<cargo_object>(cargoId)->name);
+            }
+
+            _common_format_args[0] = string_ids::buffer_1250;
+            gfx::draw_string_494BBF(*dpi, 490, dx, 118, colour::black, text_colour_id, &*_common_format_args);
+
+            dx += 10;
+            ax += 1;
+        }
+    }
+
+    // 00491A76
+    static void draw_tabs(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        auto skin = objectmgr::get<interface_skin_object>();
+        auto companyColour = companymgr::get_company_colour(window->number);
+
+        for (uint8_t stationType = 0; stationType < std::size(tabInformationByType); stationType++)
+        {
+            TabDetails tab = tabInformationByType[stationType];
+            uint32_t image = gfx::recolour(skin->img + tab.imageId, companyColour);
+            widget::draw_tab(window, dpi, image, tab.widgetIndex);
+        }
+    }
+
+    // 0x004914D8
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets and tabs.
+        window->draw(dpi);
+        draw_tabs(window, dpi);
+
+        // Draw company owner image.
+        auto company = companymgr::get(window->number);
+        auto competitor = objectmgr::get<competitor_object>(company->competitor_id);
+        uint32_t image = gfx::recolour(competitor->images[company->var_19], company->colour.primary);
+        uint16_t x = window->x + window->widgets[widx::company_select].left + 1;
+        uint16_t y = window->y + window->widgets[widx::company_select].top + 1;
+        gfx::draw_image(dpi, x, y, image);
+
+        // TODO: locale-based pluralisation.
+        _common_format_args[0] = window->var_83C == 1 ? string_ids::status_num_stations_singular : string_ids::status_num_stations_plural;
+        _common_format_args[1] = window->var_83C;
+
+        // Draw number of stations.
+        gfx::point_t origin = { (int16_t)(window->x + 4), (int16_t)(window->y + window->height - 12) };
+        gfx::draw_string_494B3F(*dpi, &origin, colour::black, string_ids::white_stringid2, &*_common_format_args);
+    }
+
+    // 0x004917BB
+    static void on_dropdown(ui::window* window, widget_index widgetIndex, int16_t itemIndex)
+    {
+        if (widgetIndex != widx::company_select)
+            return;
+
+        uint32_t companyId = dropdown::getCompanyIdFromSelection(itemIndex);
+
+        // Try to find an open station list for this company.
+        auto companyWindow = WindowManager::bringToFront(WindowType::stationList, companyId);
+        if (companyWindow != nullptr)
+            return;
+
+        // If not, we'll turn this window into a window for the company selected.
+        auto company = companymgr::get(companyId);
+        if (company->var_00 == string_ids::empty)
+            return;
+
+        window->number = companyId;
+        window->owner = companyId;
+        window->var_844 = 0;
```

Comment:
Original code doesn't seem to do this

Reply:
Just `window->var_844 = 0;`, or?

Reply:
This makes it reset the sort mode after a company switch though.